mm2src/lp_ordermatch.rs

```diff
@@ -2211,46 +2191,33 @@ pub async fn broadcast_maker_orders_keep_alive_loop(ctx: MmArc) {
         for (_, order_mutex) in my_orders {
             let order = order_mutex.lock().await;
             if let Some(p2p_privkey) = order.p2p_privkey {
-                let key_pair = match key_pair_from_secret(&p2p_privkey.0) {
-                    Ok(k) => k,
-                    Err(_) => {
-                        error!("Order {} has invalid p2p_privkey, skip P2P broadcast", order.uuid);
-                        continue;
-                    },
-                };
-                let pubsecp = hex::encode(key_pair.public_slice());
-
+                // Artem Vitae
+                // I tried if let Some(p2p_privkey) = order_mutex.lock().await.p2p_privkey
```

Comment:
Cloning of the order has to be used to drop the mutex in this case, `if let Some(p2p_privkey) = order_mutex.lock().await.clone().p2p_privkey` will work but I guess just using `drop(order)` is ok as it avoids cloning.